cql3/expr/expression.cc

```diff
@@ -2300,5 +2300,28 @@ type_of(const expression& e) {
     }, e);
 }
 
+bool is_single_column_restriction(const expression& e) {
+    if (find_in_expression<unresolved_identifier>(e, [](const auto&) {return true;})) {
+        on_internal_error(expr_logger, format("is_single_column_restriction - expression is not prepared: {}", e));
+    }
+
+    const column_value* the_column = nullptr;
+    bool result = false;
+
+    find_in_expression<column_value>(e,
+        [&](const column_value& cval) -> bool {
+            if (the_column == nullptr) {
+                the_column = &cval;
+                result = true;
+            } else if (the_column->col != cval.col) {
+                result = false;
+            }
+
+            return false;
```

Comment:
Looks unnecessarily complicated. Shouldn't you just check for binary_operator and then check that lhs is a column_value?

Reply:
This is a `single_column_restriction` in the same way that the `class single_column_restriction` was.
It can be a complex expression, but all restrictions must be on the same column.
So a restriction like:

```cql
a < 5 AND a > 0 AND a[0] IN (1, 2)
```
Would be a single column restriction.

Reply:
I see. Maybe a comment to explain it, we can't rely on the old defnition.

Reply:
There is comment it the header file